File: app/src/main/java/br/com/vicentec12/mygames/data/source/game/GameDataSource.java
package br.com.vicentec12.mygames.data.source.game;

import android.content.Context;

import androidx.annotation.NonNull;

import java.util.List;

import br.com.vicentec12.mygames.data.model.Game;
import br.com.vicentec12.mygames.interfaces.Callbacks;

public interface GameDataSource {

    interface OnGamesListedCallback {

        void onSuccess(List<Game> games);

        void onFailure();

    }

    interface OnGameGetedCallback {

        void onSuccess(Game game);

        void onFailure();

    }


    interface OnGameDeletedCallback {

        void onSuccess(int mMessage, int mNumDeleted);

        void onFailure(int mMessage);

    }

    void list(int sortBy, OnGamesListedCallback callback);

    void get(int id, OnGameGetedCallback callback);

    void insert(@NonNull Game game, Callbacks.OnLocalCallback callback);

    void update(@NonNull Game game, Callbacks.OnLocalCallback callback);

    void delete(@NonNull List<Game> games, OnGameDeletedCallback callback);

    void deleteAll();

}

File: app/src/main/java/br/com/vicentec12/mygames/data/source/game/GameDataSource.kt
package br.com.vicentec12.mygames.data.source.game

import br.com.vicentec12.mygames.data.model.Game
import br.com.vicentec12.mygames.data.source.Result

interface GameDataSource {

    suspend fun list(mConsoleId: Long, mOrderBy: Int): Result<List<Game?>>

    suspend fun get(mId: Long): Result<Game>

    suspend fun insert(mGame: Game): Result<Nothing>

    suspend fun update(mGame: Game): Result<Nothing>

    suspend fun delete(mGames: List<Game>): Result<Int>

    suspend fun deleteAll(): Result<Nothing>

}